app/main.py

```diff
@@ -1,3 +1,85 @@
+from typing import Any
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Any) -> "Distance":
+        if isinstance(other, Distance):
+            return Distance(km=self.km + other.km)
+        if isinstance(other, int | float):
+            return Distance(km=self.km + other)
+        else:
+            raise TypeError("Can't calculate")
+
+    def __iadd__(self, other: Any) -> "Distance":
+        if isinstance(other, int | float):
+            self.km += other
+            return self
+        if isinstance(other, Distance):
+            self.km += other.km
+            return self
+        else:
+            raise TypeError("Can't calculate")
```

Comment:
The exception message 'Can't calculate' is not very descriptive. Consider providing a more informative message, such as 'Unsupported operand type(s) for in-place addition'.